doc/reference/src/low-level-execution-model.pdc

```diff
@@ -111,276 +111,542 @@ The oracle-free abstract algorithm for optimal reduction operates on four node t
+    \end{tikzpicture}
+  \end{subfigure}
 \end{figure}
 
 \begin{figure}[H]
-\caption{Eraser-fan commutation}
-\begin{subfigure}[c]{0.1\textwidth}
-\begin{tikzpicture}
-\inetcell[]{a}{$f_i$}[5]
-\inode[below = 40 pt]{b}{$⊗$}
-\inwire[]{a/1}{a}
-\inwire[]{a/4}{d}
-\cutWire{a.out}{b.north}{}{}
-\end{tikzpicture}
-\end{subfigure}
-~
-\begin{subfigure}[c]{0.05\textwidth}
-\huge{→}
-\end{subfigure}
-~
-\begin{subfigure}[c]{0.1\textwidth}
-\begin{tikzpicture}
-\inode[]{a}{$⊗$}
-\inode[right = 20 pt]{b}{$⊗$}
-\inwire[]{a}{a}
-\inwire[]{b}{d}
-\end{tikzpicture}
-\end{subfigure}
+  \caption{Eraser-fan commutation}
+  \begin{subfigure}[c]{0.1\textwidth}
+    \begin{tikzpicture}
+      \inetcell[]{a}{$f_i$}[5]
+      \inode[below = 40 pt]{b}{$⊗$}
+      \inwire[]{a/1}{a}
+      \inwire[]{a/4}{d}
+      \cutWire{a.out}{b.north}{}{}
+    \end{tikzpicture}
+  \end{subfigure}
+  ~
+  \begin{subfigure}[c]{0.05\textwidth}
+    \huge{→}
+  \end{subfigure}
+  ~
+  \begin{subfigure}[c]{0.1\textwidth}
+    \begin{tikzpicture}
+      \inode[]{a}{$⊗$}
+      \inode[right = 20 pt]{b}{$⊗$}
+      \inwire[]{a}{a}
+      \inwire[]{b}{d}
+    \end{tikzpicture}
+  \end{subfigure}
 \end{figure}
 
 \begin{figure}[H]
-\caption{Eraser-eraser annihilation}
-\begin{subfigure}[c]{0.1\textwidth}
-\begin{tikzpicture}
-\inode[]{a}{$⊗$}
-\inode[below = 40 pt]{b}{$⊗$}
-\vwire{a}{b}
-\end{tikzpicture}
-\end{subfigure}
-~
-\begin{subfigure}[c]{0.05\textwidth}
-\huge{→}
-\end{subfigure}
-~
-\begin{subfigure}[c]{0.1\textwidth}
-\end{subfigure}
+  \caption{Eraser-eraser annihilation}
+  \begin{subfigure}[c]{0.1\textwidth}
+    \begin{tikzpicture}
+      \inode[]{a}{$⊗$}
+      \inode[below = 40 pt]{b}{$⊗$}
+      \vwire{a}{b}
+    \end{tikzpicture}
+  \end{subfigure}
+  ~
+  \begin{subfigure}[c]{0.05\textwidth}
+    \huge{→}
+  \end{subfigure}
+  ~
+  \begin{subfigure}[c]{0.1\textwidth}
+  \end{subfigure}
 \end{figure}
 
 \floatstyle{boxed}
 \restylefloat{figure}
 
 ### Argument for correctness
 
+
 ## Bespoke term encoding
 
 - Primitive types (integer, string, bytes) ~> node types w/data
 - n-arity primitive functions (+, -, *, concat, index) ~> rewrite rules
-- alt-term encoding: $lam x. x + x + x$ ~> $3 * x$ rewrite rule instead of multi-node graph
+- alt-term encoding: $\lambda x. x + x + x$ ~> $3 * x$ rewrite rule instead of multi-node graph
 
 - Over set of underlying opcodes & von-Neumann-ish VM (e.g. x86, LLVM, EVM).
 - Use linear types (e.g. rewrite memory locations in place)
 - Must be semantically equivalent to interaction system encoding after read-back
 - Must preserve any-order reduction (exists some maximum `n` rewrite steps after which graph read-back must have final form regardless of order)
 - Can relax diamond property (strong confluence)? unclear if necessary
 
+### Argument for Correct read-back
+
+We wish to show that a read-back of an EAL-expression that has undergone zero or more reductions gives back the same result as normal evaluation up to alpha equality.
```

Comment:
OK, so this is not only arguing for the correctness of read-back but also the correctness of reduction?

Read-back could be argued separately, e.g. "read-back (encode (term)) = term", but this is fine too.